pysite/oauth.py

```diff
@@ -0,0 +1,85 @@
+import logging
+from uuid import uuid4, uuid5
+
+from flask import session
+from flask_dance.consumer.backend import BaseBackend
+from flask_dance.contrib.discord import discord
+import requests
+
+from pysite.constants import DISCORD_API_ENDPOINT, SERVER_ID
+from pysite.mixins import DBMixin
+
+_MAIN_BACK = None  # type: pysite.database.RethinkDB
+
+
+class OauthBackend(BaseBackend, DBMixin):
+    """
+    This is the backend for the oauth
+
+    This is used to manage users that have completed
+    an oauth dance. It contains 3 functions, get, set,
+    and delete, however we only use set.
+
+    Inherits:
+        flake_dance.consumer.backend.BaseBackend
+        pysite.mixins.DBmixin
+
+    Properties:
+        key: The app's secret, we use it too make session IDs
+    """
+    table_name = "oauth_data"
+
+    def __init__(self, manager):
+        global _MAIN_BACK
+
+        super(BaseBackend, self).__init__()
+        self.setup(manager, manager.oauth_blueprint)
+        self.key = manager.app.secret_key
+        _MAIN_BACK = self
+
+    def get(self, *args, **kwargs):  # Not used
+        pass
+
+    def set(self, blueprint, token):
+        user = get_user()  # get the user's information, based on a local proxy
+        join_discord(token["access_token"], user["id"])  # Attempt to join a discord
+        sess_id = str(uuid5(uuid4(), self.key))  # Create a session ID, encrypted to protect against timing attacks
+        session["session_id"] = sess_id  # Set it
+        self.db.insert("oauth_data", {"id": sess_id, "access_token": token["access_token"], "snowflake": user['id'],
+                                      "refresh_token": token["refresh_token"], "expires_at": token["expires_at"]})
```

Comment:
Can we have something more readable here and for the insert below? Something like:
```py
self.db.insert("oauth_data", {
        "id": sess_id,
        "access_token": token["access_token"],
        "snowflake": user['id'],
        "refresh_token": token["refresh_token"],
        "expires_at": token["expires_at"]
    }
)
```